### PR TITLE
Ensure Consistent Slot Status

### PR DESCRIPTION
**Pull request by ejoerns**
So far, the slot status has only been written after a successful (not-skipped) slot update.

This makes it hard to identify what the actual content and state of the slot is.

Also, the existing `status=` field had only one possible value which was `"ok"`.

Besides some minor adaptions, the main aspect of this PR is to ensure the slot status gets updated even when the slot update failed or was skipped due to the 'slot skipping' mechanism.

In the first case, it also sets the status to `"failed"` to let the user know this slot is in an unusable state. When having skipped, it sets the state to `"skipped"` indicating it matches the content of the noted bundle information but was not explicitly written.

### CODE REVIEW

**Comment by ejoerns** [comment]
Changes:

* For slot skipping, also just mark the status as "good" since skipping should be an internal optimization of RAUC
* Ensure invalidating a slot's central status before starting writing to prevent considering a broken slot a working after a RAUC crash.

File: src/install.c [hunk 1]
```diff
@@ -997,6 +997,15 @@ static gboolean handle_slot_install_plan(const RaucManifest *manifest, const RIm
 
 		/* Dummy step to indicate slot was skipped */
 		r_context_begin_step("skip_image", "Copying image skipped", 0);
+
+		/* Update the status also for skipped slots */
+		g_message("Updating slot %s status", plan->target_slot->name);
+		update_slot_status(slot_state, "ok", manifest, plan, args);
+		if (!save_slot_status(plan->target_slot, &ierror)) {
+			g_propagate_prefixed_error(error, ierror, "Error while writing status file: ");
+			return FALSE;
```
Comment (jluebbe) [inline src/install.c:1021]: Why don't we need the `r_context_end_step` here?

File: src/install.c [hunk 2]
```diff
@@ -946,7 +946,7 @@ static gboolean pre_install_checks(gchar* bundledir, GPtrArray *install_plans, G
 	return TRUE;
 }
 
-static void update_slot_status(RaucSlotStatus *slot_state, const RaucManifest *manifest, const RImageInstallPlan *plan)
+static void update_slot_status(RaucSlotStatus *slot_state, const gchar* status, const RaucManifest *manifest, const RImageInstallPlan *plan, const RaucInstallArgs *args)
```
Comment (jluebbe) [inline src/install.c:949]: The `args` argument seems unused.

**Review by jluebbe — CHANGES_REQUESTED** [review]
Looks fine otherwise.

File: src/install.c (hunk 2 above)
Comment (ejoerns) [reply src/install.c:949]: Seems, but isn't :wink:. It is used in

``` c
slot_state->installed_txn = g_strdup(args->transaction);
```

File: src/install.c (hunk 1 above)
Comment (ejoerns) [reply src/install.c:1021]: I've checked this and would say it was just missing. The same is true for the error case of the `skip_image` step which I also fixed now.

**Comment by ejoerns** [comment]
Fixed (1st force push) and rebased onto current master (2nd force push)

**Review by jluebbe — CHANGES_REQUESTED** [review]

**Review by jluebbe — APPROVED** [review]

**Pull request merged**